### PR TITLE
Capture carb updates for create, update, and delete operations

### PR DESCRIPTION
**Pull request by darinkrauss**
- https://tidepool.atlassian.net/browse/LOOP-1417
- Update to use latest RemoteDataService.uploadCarbData
- Support new CarbStoreError error code
- Various updates to support using Result for completion handler
- Update CarbStore simulated core data
- Update CarbStoreTests

This PR is based upon changes to LoopKit in PR https://github.com/tidepool-org/LoopKit/pull/217.

Note: The next few PRs will be against the primary feature branch darinkrauss/LOOP-1417-capture-carbohydrate-entries.v2.ALL, not dev. My intention is to break the large overall feature into smaller, more easily reviewable PRs, and then merge the primary feature branch into dev after a final "summary" PR. (Several of the smaller PRs cannot be merged directly into dev because they would partially break existing functionality without a full replacement.)

Reviewers requested: 2

### CODE REVIEW

File: Loop Status Extension/StatusViewController.swift [hunk 1]
```diff
@@ -92,11 +92,11 @@ class StatusViewController: UIViewController, NCWidgetProviding {
     
     let absorptionTimes = LoopSettings.defaultCarbAbsorptionTimes
     lazy var carbStore = CarbStore(healthStore: healthStore,
-                                   observeHealthKitSamplesFromOtherApps: FeatureFlags.observeHealthKitSamplesFromOtherApps,
+                                   observeHealthKitSamplesFromOtherApps: false,
```
Comment (nhamming) [inline Loop Status Extension/StatusViewController.swift:95]: [nit] should this be hardcoded?

**Review by nhamming — APPROVED** [review]
LGTM

File: Loop Status Extension/StatusViewController.swift (hunk 1 above)
Comment (darinkrauss) [reply Loop Status Extension/StatusViewController.swift:95]: Yes, only the CarbStore for the main app needs to observe HealthKit so doing it it the status extension or watch would just be duplicating work.

Comment (darinkrauss) [reply Loop Status Extension/StatusViewController.swift:95]: Unless you can think of a use case where the status extension would be running (and able to ingest HealthKit data) where the main app would not be able to do so. I can't think of one.

**Review by ps2 — APPROVED** [review]
LGTM!

**Pull request merged**